### PR TITLE
Add missing trusted cluster info to TLS certs created at tsh login

### LINKED ISSUE #3693: tsh doesn't target the correct Kubernetes cluster if logging directly into a non-root cluster
**What happened**: I ran `tsh login --proxy=example-main.gravitational.co:3080 gus-test-cluster-seven` and logged into a Teleport cluster. This caused my `~/.tsh/profile` to get updated:

```
web_proxy_addr: example-main.gravitational.co:3080
ssh_proxy_addr: example-main.gravitational.co:3023
kube_proxy_addr: example-main.gravitational.co:3026
user: example@gravitational.com
cluster: gus-test-cluster-seven
```

Any `tsh` operation correctly targets the leaf cluster - `gus-test-cluster-seven`. Any `kubectl` operation actually targets the root cluster (`example-main.gravitational.co`) instead:

```
# shows leaf cluster, correct
$ tsh login --proxy example-main.gravitational.co gus-test-cluster-seven
If browser window does not open automatically, open it by clicking on the link:
 http://127.0.0.1:45231/92fdaadb-e589-47f6-9b18-51996768b0e1
> Profile URL:  https://example-main.gravitational.co:3080
  Logged in as: example@gravitational.com
  Cluster:      gus-test-cluster-seven
  Roles:        clusteradmin*
  Logins:       root
  Valid until:  2020-05-12 04:59:53 -0300 ADT [valid for 12h0m0s]
  Extensions:   permit-agent-forwarding, permit-port-forwarding, permit-pty


* RBAC is only available in Teleport Enterprise
  https://gravitational.com/teleport/docs/enterprise

# Teleport nodes in the leaf cluster, correct
$ tsh ls
Node Name                Address         Labels                                                                                                                            
------------------------ --------------- --------------------------------------------------------------------------------------------------------------------------------- 
ip-172-20-44-170         100.96.7.1:3022 kubelet-labels=kops.k8s.io/instancegroup=nodes|kubernetes.io/role=node|node-role.kubernetes.io/node=, node-type=kubernetes-worker 
                                         running-on=host                                                                                                                   
ip-172-20-46-62          100.96.6.0:3022 kubelet-labels=kops.k8s.io/instancegroup=nodes|kubernetes.io/role=node|node-role.kubernetes.io/node=, node-type=kubernetes-worker 
                                         running-on=host                                                                                                                   
teleport-c7bf6647d-fwrgg 127.0.0.1:3022                    

# shows leaf cluster, correct
$ kubectl config current-context
gus-test-cluster-seven

# this k8s worker is in the root cluster, not the leaf (INCORRECT)
$ kubectl -v=7 get nodes
I0511 17:01:03.752529 2447430 loader.go:375] Config loaded from file:  /home/gus/.kube/config
I0511 17:01:03.756955 2447430 round_trippers.go:420] GET https://example-main.gravitational.co:3026/api/v1/nodes?limit=500
I0511 17:01:03.756967 2447430 round_trippers.go:427] Request Headers:
I0511 17:01:03.756972 2447430 round_trippers.go:431]     User-Agent: kubectl/v1.16.2 (linux/amd64) kubernetes/c97fe50
I0511 17:01:03.756976 2447430 round_trippers.go:431]     Accept: application/json;as=Table;v=v1beta1;g=meta.k8s.io, application/json
I0511 17:01:03.891339 2447430 round_trippers.go:446] Response Status: 200 OK in 134 milliseconds
NAME                                             STATUS   ROLES    AGE    VERSION
gke-teleport-demo-cluster-pool-1-afbf7cbd-x5ck   Ready    <none>   403d   v1.14.10-gke.27
```

Rerunning the login command does not fix this. I have to explicitly switch to a different cluster (`tsh login --proxy=example-main.gravitational.co:3080 example-main.gravitational.co`) and back again (`tsh login --proxy=example-main.gravitational.co:3080 gus-test-cluster-seven`) to fix the issue. 

**What you expected to happen**: Directly logging into a cluster with `tsh login --proxy=root-cluster leaf-cluster` should correctly target all `tsh` and `kubectl` operations at the correct cluster.

**How to reproduce it (as minimally and precisely as possible)**: See above.

Manually clearing out all unrelated entries in my `~/.kube/config` doesn't seem to help. Deleting `~/.tsh/profile` doesn't help.

### Environment
- Teleport version (use `teleport version`): `Teleport Enterprise v4.2.9git:v4.2.9-0-ga4bd6c36 go1.13.2`
- Tsh version (use `tsh version`): `Teleport v4.2.9 git:v4.2.9-0-ga4bd6c36 go1.13.2` (also happens with tsh `4.2.8`)
- OS (e.g. from `/etc/os-release`): `Fedora 31`

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): Root in GKE, leaf in AWS (created by `kops`)

### PR DESCRIPTION
**Pull request by awly**
TLS credentials on login appear different from them on re-issue.
Specifically, they are missing kube users/groups in Subject, as well as
`RouteToCluster`.

With this change, user can now do:
```
$ tsh login --proxy=root-proxy leaf-proxy
$ kubectl get ...
```
and kubectl commands will get routed through `root-proxy` to `leaf-proxy`
and contain proper kube principals for the user.

Another fix bundled here is advertising `kube.listen_addr` via proxy `Ping` RPC, along with existing `kube.public_addr`. If `listen_addr` used a custom port, the client wouldn't figure it out.

Fixes #3693

### CODE REVIEW

**Comment by awly** [comment]
Reviewers: happy to chat over zoom if there's too much going on in this PR.

File: lib/client/weblogin.go
Comment (webvictim) [inline lib/client/weblogin.go:222]: As this set of values is common to all `SSHLogin*` structs, is there any way for us to define it once and reuse it? Feels like a lot of boilerplate being copy/pasted with the same comments.

File: lib/client/weblogin.go
Comment (webvictim) [inline lib/client/weblogin.go:504]: comment left in here

**Review by webvictim — COMMENTED** [review]
LGTM on the whole. Very happy to see more standardisation going on.

File: lib/client/weblogin.go
Comment (awly) [reply lib/client/weblogin.go]: Good idea, done

File: lib/client/weblogin.go
Comment (awly) [reply lib/client/weblogin.go]: Ooops, removed.

File: lib/web/apiserver.go
Comment (russjones) [inline lib/web/apiserver.go:1093]: Can the error here contain any sensitive information and is this error returned to the client? I'm inclined to log it instead of returning it in an error. That we we can be sure we're not accidentally returning sensitive information to the client.

File: lib/auth/methods.go
Comment (russjones) [inline lib/auth/methods.go:219]: Question regarding migration, is it okay for this value to be empty? If you have an old client connecting to a new server, this value will not be set.

**Review by webvictim — APPROVED** [review]
LGTM pending @russjones' comments

File: lib/auth/methods.go
Comment (awly) [reply lib/auth/methods.go:219]: Yes, if this field is empty it will not be populated in the TLS cert (in the `street` field).
Before this change, it was never populated.
Confirmed locally with newer server and older client.

File: lib/web/apiserver.go
Comment (awly) [reply lib/web/apiserver.go]: Good point. I changed this because debugging why login failed was impossible otherwise.
Changed to logging the error, it'll work for debugging too.

**Review by russjones — APPROVED** [review]
Looks good to me. Regarding the question about backporting we discussed earlier, let's hold off on backporting this since we're so close to 4.3 release. However we'll keep it as an option if we absolutely have to.

**Pull request merged**
